### PR TITLE
[css-view-transitions-2] Define the `navigation: auto` descriptor

### PR DESCRIPTION
**Pull request by noamr**
As per [resolution](https://github.com/w3c/csswg-drafts/issues/8783#issuecomment-1885359766), `auto` would match any navigation, except:
- reloads (via script, browser UI, `<meta>`, anythg)..,
- A browser-UI push/replace (e.g. typing in the URL address bar, bookmarks, etc)

Closess #8783

### CODE REVIEW

File: css-view-transitions-2/Overview.bs [hunk 1]
```diff
@@ -299,7 +299,13 @@ Note: as per default behavior, the ''@view-transition'' rule can be nested insid
 
 		: <dfn>auto</dfn>
 		:: The transition will be enabled if the navigation is same-origin, without cross-origin
-			redirects, and is not a {{NavigationType/reload}}.
+			redirects, and is either a {{NavigationType/traverse}} navigation or a navigation that's neither
```
Comment (khushalsagar) [inline css-view-transitions-2/Overview.bs:302]: Mind writing this something like:

and whose {{NavigationType}} is:
- traverse
- push or replace with {{userNavigationInvolvement}} not equal to browser UI

The sub-points make it easier to understand each case and we want to link to existing concepts in the html spec.

File: css-view-transitions-2/Overview.bs [hunk 2]
```diff
@@ -299,7 +299,13 @@ Note: as per default behavior, the ''@view-transition'' rule can be nested insid
...
+			a {{NavigationType/reload}} or initiated by browser UI.
+
+			Note: Navigations excluded from ''@view-transition/navigation/auto'' are for example, navigating
+			var the URL address bar or clicking a bookmark, as well as any form of {{NavigationType/reload}},
```
Comment (khushalsagar) [inline css-view-transitions-2/Overview.bs:306]:
```suggestion
			via the URL address bar or clicking a bookmark, as well as any form of user or script initiated {{NavigationType/reload}}
```

File: css-view-transitions-2/Overview.bs [hunk 3]
```diff
@@ -389,12 +395,14 @@ The {{CSSViewTransitionRule}} represents a ''@view-transition'' rule.
 ## Monkey patches to HTML ## {#monkey-patch-to-html}
 
 	<div algorithm="monkey patch to apply the history step">
-		Prepend a step at the beginning of the task [=queue a global task|queued=] on |navigable|'s [=active window=]
+		Prepend these steps at the beginning of the task [=queue a global task|queued=] on |navigable|'s [=active window=]
 		when <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#apply-the-history-step">applying the history step</a> (14.11.1, <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#updating-the-traversable:queue-a-global-task-3">here</a>):
 
-		This monkey-patch step assumes a boolean |changingNavigationContinuation|, a [=/navigable=] |navigable|, a {{Document}} |oldDocument|, a {{Document}} |newDocument|, and a {{NavigationType}} |navigationType|.
+		This monkey-patch step assumes a boolean |changingNavigationContinuation|, a [=/navigable=] |navigable|, a {{Document}} |oldDocument|, a {{Document}} |newDocument|, a {{NavigationType}} |navigationType|, and |userInvolvementForNavigateEvents|.
```
Comment (khushalsagar) [inline css-view-transitions-2/Overview.bs:401]: Can you cross-link to "user navigation involvement" like the other types in this list.

Comment (noamr) [reply css-view-transitions-2/Overview.bs]: OK. It's not exported correctly, but it's a monkey patch so I added a manual link.

File: css-view-transitions-2/Overview.bs (hunk 1 above)
Comment (noamr) [reply css-view-transitions-2/Overview.bs]: Done

File: css-view-transitions-2/Overview.bs [hunk 4]
```diff
@@ -299,7 +299,14 @@ Note: as per default behavior, the ''@view-transition'' rule can be nested insid
...
 		: <dfn>auto</dfn>
 		:: The transition will be enabled if the navigation is same-origin, without cross-origin
-			redirects, and is not a {{NavigationType/reload}}.
+			redirects, and whoes {{NavigationType}} is
+			* {{NavigationType/traverse}} or
+			* {{NavigationType/push}} or {{NavigationType/replace}} with <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#user-navigation-involvement">user navigation involvement</a> not equal to `"browser UI"`.
+
+			Note: Navigations excluded from ''@view-transition/navigation/auto'' are for example, navigating
+			via the URL address bar or clicking a bookmark, as well as any form of user or script initiated {{NavigationType/reload}}
+			user or script initiated.
```
Comment (khushalsagar) [inline css-view-transitions-2/Overview.bs:308]:
```suggestion
```

File: css-view-transitions-2/Overview.bs [hunk 5]
```diff
@@ -299,7 +299,14 @@ Note: as per default behavior, the ''@view-transition'' rule can be nested insid
 
 		: <dfn>auto</dfn>
 		:: The transition will be enabled if the navigation is same-origin, without cross-origin
-			redirects, and is not a {{NavigationType/reload}}.
+			redirects, and whoes {{NavigationType}} is
+			* {{NavigationType/traverse}} or
+			* {{NavigationType/push}} or {{NavigationType/replace}} with <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#user-navigation-involvement">user navigation involvement</a> not equal to `"browser UI"`.
+
+			Note: Navigations excluded from ''@view-transition/navigation/auto'' are for example, navigating
+			via the URL address bar or clicking a bookmark, as well as any form of user or script initiated {{NavigationType/reload}}
```
Comment (khushalsagar) [inline css-view-transitions-2/Overview.bs:307]:
```suggestion
			via the URL address bar or clicking a bookmark, as well as any form of user or script initiated {{NavigationType/reload}}.
```

File: css-view-transitions-2/Overview.bs [hunk 6]
```diff
@@ -389,12 +396,15 @@ The {{CSSViewTransitionRule}} represents a ''@view-transition'' rule.
...
 
 	<div algorithm="monkey patch to apply the history step">
-		Prepend a step at the beginning of the task [=queue a global task|queued=] on |navigable|'s [=active window=]
+		Prepend these steps at the beginning of the task [=queue a global task|queued=] on |navigable|'s [=active window=]
 		when <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#apply-the-history-step">applying the history step</a> (14.11.1, <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#updating-the-traversable:queue-a-global-task-3">here</a>):
 
-		This monkey-patch step assumes a boolean |changingNavigationContinuation|, a [=/navigable=] |navigable|, a {{Document}} |oldDocument|, a {{Document}} |newDocument|, and a {{NavigationType}} |navigationType|.
+		This monkey-patch step assumes a boolean |changingNavigationContinuation|, a [=/navigable=] |navigable|, a {{Document}} |oldDocument|, a {{Document}} |newDocument|, a {{NavigationType}} |navigationType|,
+		and a <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#user-navigation-involvement">user navigation involvement</a> |userInvolvementForNavigateEvents|.
```
Comment (khushalsagar) [inline css-view-transitions-2/Overview.bs:403]: Can we export "user navigation involvement" from the html spec? These steps will eventually move to the html spec but we'll still have the term referenced in the navigation descriptor definition above.

**Review by khushalsagar — APPROVED** [review]

File: css-view-transitions-2/Overview.bs (hunk 6 above)
Comment (noamr) [reply css-view-transitions-2/Overview.bs:402]: Done: https://github.com/whatwg/html/pull/10058. Will update this spec once that export is merged.

**Pull request merged**